ui/plugin-system/src/context/TimeRangeStateProvider.tsx

```diff
@@ -29,13 +29,16 @@ export function TimeRangeStateProvider(props: TimeRangeProviderProps) {
 
   const [timeRange, setActiveTimeRange] = useState<TimeRangeValue>(initialValue);
 
+  const [previousTimeRange, setPreviousTimeRange] = useState<TimeRangeValue | null>(null);
```

Comment:
IMO, this `previousTimeRange` doesn't belong in the plugin runtime sdk. "Undo" is a feature of the LineChart, therefore it should manage the previous state based on subscribing to timeRange state changes. You can argue that we should manage previous state values for everything to support "undo" use-cases otherwise.

Reply:
Yeah that's true, I had it closer to that at one point but then starting overthinking it and it's a little unclear to me what goes in the plugin runtime sdk and what doesn't. For now, I'm going to just leave the optional onDoubleClick callback undefined and come back to unzoom later since maybe it shouldn't be on double click